botocore/credentials.py

```diff
@@ -98,10 +89,24 @@ def create_credential_resolver(session, cache=None):
         instance_metadata_provider
     ]
 
+    assume_role_provider = AssumeRoleProvider(
+        load_config=lambda: session.full_config,
+        client_creator=session.create_client,
+        cache=cache,
+        profile_name=profile_name,
+        role_arn=role_arn,
+        credential_sourcer=CanonicalNameCredentialSourcer([env_provider]+providers)
```

Comment:
This is so the AssumeRole provider will attempt to assume role from whatever the otherwise default credentials are. There is a small side effect of this, though: it allows me to write 
```
[profile some-profile]
credential_source = [something that isn't Environment, Ec2InstanceMetadata, or EcsContainer], e.g. Ec2Config
```
Is this acceptable?